### PR TITLE
Remove awknowledged status & add random msg group id to sqs

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

**Review by christopherjturner — APPROVED** [review]

File: src/server/callback/listener/handle-scan-results-callback.js [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
-} from '~/src/server/common/helpers/upload-status'
 import fetch from 'node-fetch'
 
 async function handleScanResultsCallback(message, callbackQueueUrl, server) {
-  try {
-    const receiptHandle = message.ReceiptHandle
-    const payload = JSON.parse(message.Body)
-    const uploadDetails = await server.redis.findUploadWithFiles(
-      payload.uploadId
+  const receiptHandle = message.ReceiptHandle
```
Comment (feedmypixel) [inline src/server/callback/listener/handle-scan-results-callback.js:6]: This I feel is more preference. Theres a lot of if/else/else if statements in here that make it hard to follow. A few potential ways of making this easier to follow:

- Just use `if`, don't nest conditionals, so avoid using `else` or `else if` statements
- Abstract things out to local functions so its simple to see what a body of work is doing by the name of the `function/method`

File: src/server/common/helpers/redis-helper.js [hunk 2]
```diff
@@ -29,25 +29,27 @@ class RedisHelper {
...
-        this.updateField(result.fields, fileId, {
-          s3Key: file?.s3Key,
-          s3Bucket: file?.s3Bucket,
-          fileStatus: file?.fileStatus
-        })
+  async findUploadAndFiles(uploadId) {
+    const uploadDetails = await this.findUploadDetails(uploadId)
+    const files = []
+
+    if (uploadDetails && uploadDetails.fileIds) {
```
Comment (feedmypixel) [inline src/server/common/helpers/redis-helper.js:36]: could be `if (uploadDetails?.fileIds) {`

File: src/server/common/helpers/redis-helper.js [hunk 3]
```diff
@@ -29,25 +29,27 @@ class RedisHelper {
...
+      for (const fileId of uploadDetails.fileIds) {
```
Comment (feedmypixel) [inline src/server/common/helpers/redis-helper.js:37]: I wonder if building up the array of promises and using `Promise.all` here would be more efficient?. Or maybe even using `Promise.allSettled()` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

File: src/server/common/helpers/redis-helper.js [hunk 4]
```diff
@@ -58,14 +60,12 @@ class RedisHelper {
       return false
     }
 
-    for (const key of Object.keys(root)) {
-      const value = root[key]
-
+    for (const value of Object.values(root)) {
```
Comment (feedmypixel) [inline src/server/common/helpers/redis-helper.js:63]: just preference but could `.map()` or `.reduce()` by used here and below?

File: src/server/scan/listener/helpers/process-scan-complete.js [hunk 5]
```diff
@@ -8,28 +9,39 @@ import { config } from '~/src/config'
...
 const callbackQueue = config.get('sqsScanResultsCallback')
 async function processScanComplete(server, uploadId) {
-  const upload = await server.redis.findUploadWithFiles(uploadId)
-  if (isUploadPending(upload.uploadStatus) && isScanningComplete(upload)) {
-    upload.uploadStatus = uploadStatus.ready.description
-    upload.numberOfInfectedFiles = numberOfInfectedFiles(upload)
-    await server.redis.storeUploadDetails(uploadId, upload)
-    if (upload.scanResultCallbackUrl) {
+  const { files, uploadDetails } =
+    await server.redis.findUploadAndFiles(uploadId)
```
Comment (feedmypixel) [inline src/server/scan/listener/helpers/process-scan-complete.js:13]: so if there is no result here then this will blow up. Maybe guard this:

```
const uploadAndFilesResult = await server.redis.findUploadAndFiles(uploadId)
const files = uploadAndFilesResult?.files
const uploadDetails = uploadAndFilesResult?.uploadDetails

```

File: src/server/scan/listener/helpers/process-scan-complete.js [hunk 6]
```diff
@@ -8,28 +9,39 @@ import { config } from '~/src/config'
...
+    await server.redis.storeUploadDetails(uploadId, uploadDetails)
+    if (uploadDetails.scanResultCallbackUrl) {
       await sendSqsMessage(server.sqs, callbackQueue, { uploadId })
     }
-    server.logger.info(`uploadId ${uploadId} has been marked as ready`)
+    server.logger.info(
+      uploadDetails,
+      `uploadId ${uploadId} has been marked as ready`
+    )
+  } else if (isReady(uploadDetails.ready)) {
```
Comment (feedmypixel) [inline src/server/scan/listener/helpers/process-scan-complete.js:30]: would this be clearer if its just an `if`?

**Review by feedmypixel — COMMENTED** [review]
A few comments - mostly just preference, some things  could be guarded better and I wonder if there are simpler ways to show how this code works and make it a little easier to grok?

Also worth having a look at the `Promise.all/allsettled` to provide concurrency, rather than using `for of` to loop async actions.
